src/StringParser/Combinators.purs

```diff
@@ -142,11 +142,9 @@ sepEndBy p sep = (sepEndBy1 p sep <#> NEL.toList) <|> (sep $> Nil) <|> pure Nil
 sepEndBy1 :: forall a sep. Parser a -> Parser sep -> Parser (NonEmptyList a)
 sepEndBy1 p sep = do
   a <- p
-  ( do
-      _ <- sep
-      as <- sepEndBy p sep
-      pure (cons' a as)
-  ) <|> pure (NEL.singleton a)
+  as <- many $ try (sep *> p)
```

Comment:
If I'm not mistaken, this changes the behavior of the function in case of a parser error in `p` due to the usage of `try`. PS I'm currently on vacation and may not be very swift to respond.

Reply:
What you pointed out is correct.
Thanks for the quick review.

Reply:
NOTE

It gets
```purs
> runParser (sepEndBy (char 'a') (char ';')) "a;b"
(Right ('a' : Nil))
```
It seems that b is trashed. I think this is a little problem.

I found this result occurs also in original `sebEndBy`.
I wonder if it is the intended behavior.

Reply:
Hm yeah, I can't think of obvious situations where this would be what you want. In test/BasicSpecs.purs we have some basic specs that say what should be parsable and what shouldn't. Perhaps we should extend these specs to also cover partial successes and assert on the expected remaining unparsed string. 

We should also look at what purescript-parsing does in these cases.

Reply:
I would naievely expect sepEndBy to successfully parse a separated sequence and whenever either the `sep` or the `p` fails to just stop parsing and yield what has been parsed so far.